cc_utilities/common.py

```diff
@@ -176,13 +180,9 @@ def upload_data_to_commcare(
         search_field=search_field,
         create_new_cases=create_new_cases,
     )
-    with requests.Session() as session:
```

Comment:
I'm not too familiar with Sessions, but since this used to be in a context-manager and now isn't, is there any need to "close" the session when complete?

Reply:
Hm. I've seen it used both ways:

https://requests.readthedocs.io/en/master/user/advanced/#session-objects

I can't think of a reason that we would need to make sure the session is closed at a specific time (I'm pretty sure Python will close it eventually). @benjaminEwhite do you happen to recall if there's a reason it was done this way?

Reply:
Unfortunately, I don't have much additional perspective here. That was me following what I saw in the docs for requests.  The docs do say:

> This will make sure the session is closed as soon as the with block is exited, even if unhandled exceptions occurred.

Which makes it sound like maybe??? the session won't be closed if there's an unhandled exception.

Reply:
😂 Got it, thanks!

I think that's probably okay for us since it will certainly be "closed" when the script dies. Maybe that's more relevant for persistent processes where such cruft could accumulate over time, eating up TCP connections, etc?